### DISCUSSION TITLE
Provide Ghostty configuration and documentation website as ghostty.org/documentation/llms-full.txt

### DISCUSSION THREAD

**Discussion by dsp** [ghostty-org/ghostty#3058] [Feature Requests, Ideas]
I understand this might be slightly contentious, but I find it very helpful to have websites host an `llms-full.txt` that contains the full documentation as one big text file. This is aimed for easy ingesting to LLMs. This makes it easy to ask an LLM to ask and cite for a particular keybindings, etc.

If accepted as an issue, I would go concatenate the documentation mdx files during build into llms-full.txt. In addition I would probably go ahead and daily build ghostty and output `ghostty +show-config --default --docs` and create a PR for the documentation website, such that we keep it up to date, or find another way to generate the documentation in text form.

**Comment by mitchellh** [comment]
> I would probably go ahead and daily build ghostty and output ghostty +show-config --default --docs and create a PR for the documentation website, such that we keep it up to date, or find another way to generate the documentation in text form.

The config and keybindings reference data is already auto-generated. `zig build -Demit-webdata`

> I understand this might be slightly contentious, but I find it very helpful to have websites host an llms-full.txt that contains the full documentation as one big text file. This is aimed for easy ingesting to LLMs. This makes it easy to ask an LLM to ask and cite for a particular keybindings, etc.

Is there some background on this? Aren't agent-capable LLMs now able to query documentation pages pretty well live on the internet? I'm curious if we have to do this at all...

**Reply by dsp** [reply]
> Is there some background on this? Aren't agent-capable LLMs now able to query documentation pages pretty well live on the internet? I'm curious if we have to do this at all...

ChatGPT, Claude and others currently cannot do this. I think, likely in 12 months they will be able to use agentic behavior to traverse documentation. This would likely also mostly work with LLMs that can do tool use very well, which few open source ones can do well. `llms-full.txt` is a helper for people to download a file that they can pass on to an LLM. It's a simplification aimed at the current limitations of LLMs. I very much expect this to not be used in 12 months from now, but for now it would be very useful, particularly for first time users who might want to rely on ChatGPT, Claude, Gemini, Ollama & Co
